### PR TITLE
EP-Launch Lite: correctly find newest version 

### LINKED ISSUE #9383: EP-Launch Lite fail to pickup 22-1-0 as newer than 9-6-0
Issue overview
--------------

I have both 22-1-0 and 9-6-0 installed, and EP-LaunchLite finds 9-6-0

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): mac
 - Version of EnergyPlus (if using an intermediate build, include SHA): 22.1.0

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

- Fixes #9383
- Need to sort versions properly, fails to pickup '22-1-0' as newer than '9-6-0' (it sorts strings alphabetically and '2' < '6')

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: third_party/EP-Launch-Lite/EPLaunchLite/EnergyPlusPath.py [hunk 1]
```diff
@@ -31,7 +31,10 @@ def get_latest_eplus_version():
         install_folders = glob.glob('/Applications/EnergyPlus*')
 
         # then process them into a nice list
-        ep_versions = sorted([EnergyPlusPath.get_version_number_from_path(x) for x in install_folders])
+        ep_versions = [EnergyPlusPath.get_version_number_from_path(x)
+                       for x in install_folders]
+        ep_versions = sorted(
+            ep_versions, key=lambda x: [int(x) for x in x.split('-')])
```
Comment (jmarrec) [inline third_party/EP-Launch-Lite/EPLaunchLite/EnergyPlusPath.py:37]: Until now it worked:

```python
In [5]: ep_versions = ['9-4-0', '9-6-0', '9-5-0']

In [6]: sorted(ep_versions)
Out[6]: ['9-4-0', '9-5-0', '9-6-0']

In [7]: sorted(ep_versions, key=lambda x: [int(x) for x in x.split('-')])
Out[7]: ['9-4-0', '9-5-0', '9-6-0']
```

Now with 22-1-0 in the mix, it fails:

```python
In [8]: ep_versions = ['9-4-0', '9-6-0', '22-1-0', '9-5-0']

In [9]: sorted(ep_versions)
Out[9]: ['22-1-0', '9-4-0', '9-5-0', '9-6-0']

In [10]: sorted(ep_versions, key=lambda x: [int(x) for x in x.split('-')])
Out[10]: ['9-4-0', '9-5-0', '9-6-0', '22-1-0']
```

**Comment by Myoldmopar** [comment]
I'm having trouble building this into an app bundle.  This code is still using the Python2 based Gtk code.  I think I need to make a quick pass over it to get it to Python3 gobject structure.  It's generally not a big effort to make that transition.  I'll put that on the list since it should be a small, fun, diversion, which will enable this PR to get closer to complete.

**Comment by nrel-bot-3** [comment]
@jmarrec @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@jmarrec @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@jmarrec @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
@jmarrec if you feel it's worth pushing on this version of EP-Launch Lite, could you comment here to confirm?  I'm inclined to say let's push forward into more modern libraries as I've done here: https://github.com/Myoldmopar/EPLaunchLight/pull/2.  I think either way we need to push on it a little to wrap up packaging, so I'm curious which way you'd like to go.  Thanks!

**Comment by jmarrec** [comment]
@Myoldmopar I don't feel strongly (this isn't even a pivotal ticket, I'm just hot fixing something I saw)

I feel like this should be easy to rebuild, but apparently it's not. My main issue is that I don't have an Intel mac on hand to test the app bundle myself (and I just CANNOT use python 2 on an M1 mac, as it won't build at all, while on an Intel mac you could).

If you have a more modern version *that's ready to drop*, then that's probably better. We still want to make sure that code change is also in the new codebase.

If packaging that new version isn't going to happen by the next release, then the hot fix of the old version is worth it IMHO.

**Comment by nrel-bot-3** [comment]
@Myoldmopar @jmarrec @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
I have made multiple tries to get this stuff moving.  Pyinstaller is struggling to build a running package, and I rebuilt the entire tool, first with an updated GObject, and then moving to Tk.  Everything runs happily except when trying to run it as an App bundle.  (Running the underlying binary works perfectly fine, so Pyinstaller is properly grabbing all the pieces needed, it's weird.)  Anyway, I think we'll just have to hold off on this and try to fix it after release.  If we can magically get EPLaunchLite working, we've still got a couple days to try to get it inside 22.2.

**Comment by nrel-bot-3** [comment]
@Myoldmopar @jmarrec @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@Myoldmopar @jmarrec @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar @jmarrec @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@Myoldmopar @jmarrec @myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@Myoldmopar @jmarrec it has been 22 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar @jmarrec it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar @jmarrec it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@Myoldmopar @jmarrec it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar @jmarrec it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar @jmarrec it has been 7 days since this pull request was last updated.

**Comment by jmarrec** [comment]
@Myoldmopar I'm getting tired of receiving the weekly `it has been 7 days since this pull request was last updated.`.

Do we close it or should I find an x86_64 mac (or even just use a damn github actions runner) to do it?

**Comment by jmarrec** [comment]
@Myoldmopar I tried getting the CI via github actions to build it. But there doesn't seem to be a decently simple-enough way to install Pygobject / Gtk for python2 on mac. The brew formulas were removed. I've tried macports, but it's maybe still not findng cairo and co: https://github.com/jmarrec/EPLaunchLight/actions/runs/5099363382/jobs/9167009235#step:7:17

Since I don't have a way to test them, not sure.

**Comment by nrel-bot-2c** [comment]
@Myoldmopar @jmarrec it has been 7 days since this pull request was last updated.

**Pull request closed without merging**